File: edudoc/src/app/model/interfaces/custom/progress-report.dto.ts
import { IEntity } from '@mt-ng2/base-service';
import { IProgressReport } from '../progress-report';
import { IDistrictProgressReportDate } from '../district-progress-report-date';
import { QuarterEnum } from '@model/enums/progress-report-quarters.enum';

export interface IProgressReportDto extends IEntity {
    Id: number;
    StudentId: number;
    FirstName: string;
    LastName: string;
    DateRanges: IDistrictProgressReportDate;
    FirstQuarterProgressReports: IProgressReport[];
    SecondQuarterProgressReports: IProgressReport[];
    ThirdQuarterProgressReports: IProgressReport[];
    FourthQuarterProgressReports: IProgressReport[];
    PreviousProgressReports: IProgressReport[];
    ProgressReports: IProgressReport[];
    Quarters: number[];
    SupervisorId: number;
}


export interface IProgressReportDtoWithHelpers extends IProgressReportDto {
    UnnecessaryQuarters: {[Q in QuarterEnum]: boolean},
    CompletedQuarters: {[Q in QuarterEnum]: boolean},
}

File: edudoc/src/Model/DTOs/ProgressReportDTO.cs
using System.Data.SqlClient;
using System.Data.SqlClient;
using System;
using System.Collections.Generic;

namespace Model.DTOs
{
    public class ProgressReportDto
    {
        public int StudentId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DistrictProgressReportDate DateRanges {get;set;}
        public IEnumerable<ProgressReport> FirstQuarterProgressReports { get; set; }
        public IEnumerable<ProgressReport> SecondQuarterProgressReports { get; set; }
        public IEnumerable<ProgressReport> ThirdQuarterProgressReports { get; set; }
        public IEnumerable<ProgressReport> FourthQuarterProgressReports { get; set; }
        public IEnumerable<ProgressReport> PreviousProgressReports { get; set; }
        public IEnumerable<ProgressReport> ProgressReports { get; set; }
        public IEnumerable<int> Quarters {get; set; }
        public int SupervisorId { get; set; }
    }
}
